rewrite-gradle/src/test/java/org/openrewrite/gradle/AddDependencyTest.java

```diff
@@ -1345,6 +1341,240 @@ void addUnconditionally() {
+                    }
+                    """,
+                  spec -> spec.path("build.gradle")
+                )));
+        }
+
+        @Test
+        void actNormalForDefaultTestWithSuiteDefinition() {
+            rewriteRun(
+              spec -> spec.recipe(addDependency("com.google.guava:guava:29.0-jre")),
+              mavenProject("project",
+                srcMainJava(
+                  java(usingGuavaIntMath, sourceSpecs -> sourceSet(sourceSpecs, "test"))),
+                buildGradle(
+                  //language=groovy
+                  """
+                    plugins {
+                        id "java-library"
+                        id 'jvm-test-suite'
+                    }
+                    
+                    repositories {
+                        mavenCentral()
+                    }
+                    
+                    dependencies {
+                        implementation "org.apache.logging.log4j:log4j-core:2.22.1"
+                    }
+                    
+                    testing {
+                        suites {
+                            test(JvmTestSuite) {
+                            }
+                        }
+                    }
+                    """,
+                  """
+                    plugins {
+                        id "java-library"
+                        id 'jvm-test-suite'
+                    }
+                    
+                    repositories {
+                        mavenCentral()
+                    }
+                    
+                    dependencies {
+                        implementation "org.apache.logging.log4j:log4j-core:2.22.1"
+                        
+                        testImplementation "com.google.guava:guava:29.0-jre"
+                    }
+                    
+                    testing {
+                        suites {
+                            test(JvmTestSuite) {
+                            }
+                        }
+                    }
+                    """,
+                  spec -> spec.path("build.gradle")
+                )));
+        }
+
+        @Test
+        void actNormalForSuiteWithoutDependencyBlock() {
+            rewriteRun(
+              spec -> spec.recipe(addDependency("com.google.guava:guava:29.0-jre")),
+              mavenProject("project",
+                srcMainJava(
+                  java(usingGuavaIntMath, sourceSpecs -> sourceSet(sourceSpecs, "integrationTest"))),
+                buildGradle(
+                  //language=groovy
+                  """
+                    plugins {
+                        id "java-library"
+                        id 'jvm-test-suite'
+                    }
+                    
+                    repositories {
+                        mavenCentral()
+                    }
+                    
+                    testing {
+                        suites {
+                            integrationTest(JvmTestSuite) {
```

Comment:
A test suite that lacks dependencies originally is perfectly valid. I would think if we have a collision, then we'd probably add the dependencies block for the not present case.

Reply:
So you say it's more convenient to add dependencies to the `testing.suites.XYZ.dependencies` block than adding it to the global one?
If so, we have to expand the `AddDependencyVisitor`s logic.

Reply:
This is the one where I was saying the issue is more complex and we may choose to solve this with a future PR.

In my mind, I'm trying to think about what would the user expect to happen and then we would align to that. I do agree that it's certainly easier to put the dependency at the root. And furthermore, Gradle doesn't provide any strong recommendation that I could find on using one versus the other.

For now though, I think the logic that is present is probably good enough to progress with, but I just wanted to mention it here as well.